dll/shellext/netshell/lanconnectui.cpp

```diff
@@ -17,6 +17,10 @@ CNetConnectionPropertyUi::~CNetConnectionPropertyUi()
     if (m_pNCfg)
         m_pNCfg->Uninitialize();
 
+    // Note: MSDN says we can only unlock after INetCfg::Uninitialize
```

Comment:
Why not. Since the lock has been acquired before the initialize call.

Reply:
Don't ask me, this is what MSDN says.

You must take the lock before Initialize and release it after Uninitialize.

Reply:
The lock can't be acquired or released if `bInitialized` is true, according to the code behind this.